File: Interview Problems/Parking Lot/Java Code/domain/Floor.java
package domain;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class Floor {

    private final UUID id;
    private final int floorNumber;
    private final List<ParkingSlot> slots;

    public Floor(int floorNumber) {
        this.id = UUID.randomUUID();
        this.floorNumber = floorNumber;
        this.slots = new ArrayList<>();
    }

    public void addSlot(ParkingSlot slot) {
        slots.add(slot);
    }

    public List<ParkingSlot> getAvailableSlots(Vehicle.VehicleType type) {
        List<ParkingSlot> available = new ArrayList<>();
        for (ParkingSlot slot : slots) {
            if (slot.getSlotType() == type && !slot.isOccupied()) {
                available.add(slot);
            }
        }
        return available;
    }

    public UUID getId() { return id; }
    public int getFloorNumber() { return floorNumber; }
    public List<ParkingSlot> getSlots() { return slots; }

    @Override
    public String toString() {
        return "Floor{id=" + id + ", number=" + floorNumber + ", slots=" + slots.size() + "}";
    }
}

File: Interview Problems/Parking Lot/Python Code/domain/floor.py
import uuid
from typing import List
from domain.vehicle import VehicleType
from domain.parking_slot import ParkingSlot

class Floor:
    def __init__(self, floor_number: int):
        self.id = uuid.uuid4()
        self.floor_number = floor_number
        self.slots: List[ParkingSlot] = []

    def add_slot(self, slot: ParkingSlot):
        self.slots.append(slot)

    def get_available_slots(self, vehicle_type: VehicleType) -> List[ParkingSlot]:
        return [slot for slot in self.slots if slot.slot_type == vehicle_type and not slot.is_occupied]

    def __repr__(self):
        return f"Floor(id={self.id}, number={self.floor_number}, slots={len(self.slots)})"
